# Le Sample

This tuto shows you how to get the number of sample needed per variation to reach a statistical significant result.  
First, let's import the right tools :

In [1]:
from leab import before

As we want to know the sample needed, we stand **before** launching the AB test.   
Then, from the `before` import, we create a `leSample` instance that we call `ab_test`.  
In this example we want to get a volume of sample per variation needed if we expect a minimum detectable effect of 2% on a baseline conversion rate of 20% :

In [2]:
ab_test = before.leSample(conversion_rate=20, 
                          min_detectable_effect=2)
ab_test.get_size_per_variation()

6347

So the result is `6347` visits per variation (`6347` for the A variation and `6347` more for the B variation).  
If we know the trafic expected on this test, we can ends up with a approximate number of days needed to run the test :

In [3]:
ab_test.get_duration(avg_daily_total_visitor=1000)

13

You will need `13` days to run the test.  
Then you can even generate a clear report :

In [ ]:
ab_test.report()

A `html` file is saved on the spot.